bigchaindb/backend/query.py

```diff
@@ -372,3 +372,17 @@ def get_validator_set(conn, height):
     """
 
     raise NotImplementedError
+
+
+@singledispatch
+def get_asset_tokens_for_public_keys(connection, asset_id,
```

Comment:
When do we need to query for multiple `public_keys`?

I see you use this function to extract all the votes, so the `operation` is in a way implicit with the `asset_id`.

I'm wondering if we can simplify it using a single `public_key` and removing `operation`.

Reply:
Are you suggesting someting as follows?

```python
def get_asset_tokens_for_public_key(conn, asset_id, public_key):
    query = {'outputs.public_keys': [public_key],
             'asset.id': asset_id}

    cursor = conn.run(
        conn.collection('transactions').aggregate([
            {'$match': query},
            {'$project': {'_id': False}}
        ]))
    return cursor
```

Reply:
Yes, it seems simpler and easier to use.